scripts/update-schema-id

```diff
@@ -0,0 +1,245 @@
+#
+#     ex -s -c '/^$id:/s/^.*$/$id: '\''https:\/\/spec.openapis.org\/oas\/3.1\/schema\/2024-05-07'\''/' \
+#           -c wq schemas/v3.1/schema.yaml
+#
+#     ./scripts/yaml2json/yaml2json.js schemas/v3.1/schema.yaml
+#
+#     ex -s -c '/^$id:/s/^.*$/$id: '\''https:\/\/spec.openapis.org\/oas\/3.1\/schema-base\/2024-05-07'\''/' \
+#           -c '/^$ref:/s/^.*$/$ref: '\''https:\/\/spec.openapis.org\/oas\/3.1\/schema\/2024-05-07'\''/' \
+#           -c wq schemas/v3.1/schema-base.yaml
+#
+#     ./scripts/yaml2json/yaml2json.js schemas/v3.1/schema-base.yaml
+#
+
+
+require 'bundler/inline'
+require 'time'
+
+gemfile do
+  source 'https://rubygems.org'
+  gem 'optparse'
+end
+
+
+#------------------------------------------------------------
+# main
+#------------------------------------------------------------
+def main(args)
+  # Process the command line options.
+  options = Options.process(args)
+
+  # Update schemas/v{major}.{minor}/schema.yaml.
+  update_schema_yaml(options)
+
+  # Generate schemas/v{major}.{minor}/schema.json from the updated YAML file.
+  update_schema_json(options)
+
+  # Update schemas/v{major}.{minor}/schema-base.yaml.
+  update_schema_base_yaml(options)
+
+  # Generate schemas/v{major}.{minor}/schema-base.json from the updated YAML file.
+  update_schema_base_json(options)
+end
+
+
+#------------------------------------------------------------
+# Ver
+#------------------------------------------------------------
+class Ver
+  attr_reader :major, :minor
+
+  def initialize(version)
+    m = /^(\d+)\.(\d+)$/.match(version)
+
+    @major = m[1]
+    @minor = m[2]
+  end
+end
+
+
+#------------------------------------------------------------
+# Options
+#------------------------------------------------------------
+class Options < OptionParser
+  DESC_VERSION = "specifies the target version in the 'major.minor' format. For example, '3.1'. This option is mandatory."
+  DESC_DATE    = "specifies the release date in the 'YYYY-MM-DD' format. For example, '2024-10-01'. If omitted, the current date is used."
+  DESC_DRY_RUN = "turns on the dry-run mode that shows external commands to execute but does not actually execute them."
+  DESC_QUIET   = "turns on the quiet mode."
+
+  attr_reader :version, :date, :dry_run, :quiet
+
+  def initialize
+    super
+
+    @version = nil
+    @date    = Time.now.utc.strftime('%F')
+    @dry_run = false
+    @quiet   = false
+
+    self.on('-v VERSION', '--version VERSION', DESC_VERSION) do |value|
+      check_pattern(value, 'version', /^\d+\.\d+$/)
+      @version = Ver.new(value)
+    end
+
+    self.on('-d DATE', '--date DATE', DESC_DATE) do |value|
+      check_pattern(value, 'date', /^\d{4}-\d{2}-\d{2}$/)
+      @date = value
+    end
+
+    self.on('-n', '--dry-run', DESC_DRY_RUN) do
+      @dry_run = true
+    end
+
+    self.on('-q', '--quiet', DESC_QUIET) do
+      @quiet = true
+    end
+  end
+
+  private
+
+  def error_exit(message)
+    $stderr.puts message
+    exit false
+  end
```

Comment:
built-in `abort` (method of Kernel) does the same as this `error_exit` method

Reply:
@notEthan Thank you! I have replaced `error_exit` with `abort`.